### PR TITLE
migration-sdk-style-prjs-remap-assembly-ref

### PR DESCRIPTION
**Pull request by moljac**
remap-assembly-ref migrated

### CODE REVIEW

File: build-tools/xaprepare/xaprepare/Application/Utilities.cs [hunk 1]
```diff
@@ -99,7 +99,10 @@ public static async Task<bool> BuildRemapRef (Context context, bool haveManagedR
 			bool result = await msbuild.Run (
 				projectPath: projectPath,
 				logTag: "remap-assembly-ref",
-				binlogName: "build-remap-assembly-ref"
+				arguments: new List<string> {
+					"/restore",
```
Comment (grendello) [inline build-tools/xaprepare/xaprepare/Application/Utilities.cs:103]: I think it's better to add `/restore` to `MSBuildRunner.StandardArguments`, seeing that we will be needing the parameter everywhere.

File: build-tools/xaprepare/xaprepare/Steps/Step_BuildMonoRuntimes.Unix.cs [hunk 2]
```diff
@@ -88,6 +88,9 @@ async Task<bool> ConjureXamarinCecilAndRemapRef (Context context, bool haveManag
 			string projectPath = Path.Combine (Configurables.Paths.BuildToolsDir, "conjure-xamarin-android-cecil", "conjure-xamarin-android-cecil.csproj");
 			result = await msbuild.Run (
 				projectPath: projectPath,
+				arguments: new List<string> {
+					"/restore",
```
Comment (grendello) [inline build-tools/xaprepare/xaprepare/Steps/Step_BuildMonoRuntimes.Unix.cs:92]: Ditto here

File: build-tools/xaprepare/xaprepare/Steps/Step_InstallMonoRuntimes.cs [hunk 3]
```diff
@@ -97,6 +97,9 @@ async Task<bool> ConjureXamarinCecilAndRemapRef (Context context, bool haveManag
 			result = await msbuild.Run (
 				projectPath: projectPath,
 				logTag: "conjure-xamarin-android-cecil",
+				arguments: new List<string> {
+					"/restore",
```
Comment (grendello) [inline build-tools/xaprepare/xaprepare/Steps/Step_InstallMonoRuntimes.cs:101]: And ditto here

File: build-tools/xaprepare/xaprepare/ToolRunners/MSBuildRunner.cs [hunk 4]
```diff
@@ -34,6 +34,7 @@ public async Task<bool> Run (string projectPath, string logTag, List<string>? ar
 			AddArguments (runner, StandardArguments);
 			if (!String.IsNullOrEmpty (binlogName)) {
 				string logPath = Utilities.GetRelativePath (workingDirectory!, Path.Combine (Configurables.Paths.BuildBinDir, $"msbuild-{Context.BuildTimeStamp}-{binlogName}.binlog"));
+				runner.AddQuotedArgument ("/restore");
```
Comment (grendello) [inline build-tools/xaprepare/xaprepare/ToolRunners/MSBuildRunner.cs:37]: `/restore` should be added to `StandardArguments` in the constructor. It should be harmless even if it's repeated.

**Review by grendello — CHANGES_REQUESTED** [review]

**Review by grendello — APPROVED** [review]

File: build-tools/xaprepare/xaprepare/Application/Utilities.cs [hunk 5]
```diff
@@ -99,7 +99,7 @@ public static async Task<bool> BuildRemapRef (Context context, bool haveManagedR
 			bool result = await msbuild.Run (
 				projectPath: projectPath,
 				logTag: "remap-assembly-ref",
-				binlogName: "build-remap-assembly-ref"
+				binlogName: "build-remap-assembly-ref"
```
Comment (radekdoulik) [inline build-tools/xaprepare/xaprepare/Application/Utilities.cs:102]: Nitpik white space :-)

Comment (moljac) [reply build-tools/xaprepare/xaprepare/Application/Utilities.cs]: will fix

**Pull request merged**